### PR TITLE
Remove leftover references to setValue

### PR DESCRIPTION
**Pull request by jenn-le**
Removes remaining setValue references.

### CODE REVIEW

File: experimental/dds/tree2/src/test/objMerge.ts [hunk 1]
```diff
@@ -45,13 +45,6 @@ export function merge<T>(lhs: T, rhs: T): Conflicted | Conflict | T {
 		return new Conflict(lhs, rhs);
 	}
 
-	// null is of type object, and needs to be treated as distinct from the empty object.
-	// Handling it early also avoids type errors trying to access its keys.
-	// Rationale: 'undefined' payloads are reserved for future use (see 'SetValue' interface).
-	if (lhs === null || rhs === null) {
-		return new Conflict(lhs, rhs);
-	}
-
```
Comment (jenn-le) [inline experimental/dds/tree2/src/test/objMerge.ts:54]: @yann-achard-MS I removed this so I could put a comment on it. What should we do about this?

File: experimental/dds/tree2/src/feature-libraries/forestRepairDataStore.ts [hunk 2]
```diff
@@ -119,19 +119,6 @@ export class ForestRepairDataStore<TChange> implements RepairDataStore<TChange>
...
-			// `undefined` as opposed to leaving the value unchanged.
-			if (Object.prototype.hasOwnProperty.call(modify, "setValue")) {
-				if (node.data === undefined) {
-					node.data = repairDataFactory();
-				}
-				const value = cursor.value;
-				if (node.data.value === undefined) {
-					node.data.value = new Map();
-				}
-				node.data.value.set(revision, value);
```
Comment (yann-achard-MS) [inline experimental/dds/tree2/src/feature-libraries/forestRepairDataStore.ts:133]: You can also remove the `RepairData.value` value field. Don't bother refactoring `RepairData` though.

File: experimental/dds/tree2/src/test/feature-libraries/sequence-field/sequenceChangeRebaser.spec.ts [hunk 3]
```diff
@@ -55,7 +55,6 @@ function generateLineage(
 }
 
 const testChanges: [string, (index: number, maxIndex: number) => SF.Changeset<TestChange>][] = [
-	["SetValue", (i) => Change.modify(i, TestChange.mint([], 1))],
```
Comment (yann-achard-MS) [inline experimental/dds/tree2/src/test/feature-libraries/sequence-field/sequenceChangeRebaser.spec.ts:58]: This effectively tests nested changes, so we want to keep it. Just rename it to "NestedChange".

File: experimental/dds/tree2/src/test/objMerge.ts (hunk 1 above)
Comment (yann-achard-MS) [reply experimental/dds/tree2/src/test/objMerge.ts:54]: You can just remove the "Rationale" line. It doesn't make much sense in this context anymore.

**Review by yann-achard-MS — APPROVED** [review]

**Pull request merged**